scylla-gdb.py

```diff
@@ -589,8 +589,45 @@ def __bool__(self):
         return self.__nonzero__()
 
 
+class absl_container:
+    def __init__(self, ref):
+        self.val = ref
+        self.size = ref["settings_"]["value"]["compressed_tuple_"]["value"]
+
+    def __len__(self):
+        return self.size
+
+    def __iter__(self):
+        if self.size == 0:
+            return
+        capacity = int(self.val["settings_"]["value"]["capacity_"])
+        control = self.val["settings_"]["value"]["control_"]
+        slot_type = lookup_type(str(self.val.type.strip_typedefs()) + "::slot_type").strip_typedefs()
+        for item in range(capacity):
+            ctrl_t = int(control[item])
+            # if the control is empty or deleted, its value is less than -1, see
+            # https://github.com/abseil/abseil-cpp/blob/c1e1b47d989978cde8c5a2a219df425b785a0c47/absl/container/internal/raw_hash_set.h#L487-L503
+            if ctrl_t == -1:
+                break
+            if ctrl_t >= 0:
+                yield self.val["settings_"]["value"]["slots_"].cast(slot_type.pointer())[item]
+
+    def __nonzero__(self):
+        return self.size > 0
+
+    def __bool__(self):
+        return self.size > 0
+
+
 def unordered_map(ref):
-    return flat_hash_map(ref) if ref.type.name.startswith('flat_hash_map') else std_unordered_map(ref)
+    if ref.type.name.startswith('flat_hash_map'):
+        if 'slots_' in ref:
+            return flat_hash_map(ref)
+        else:
+            # newer absl container uses a different memory layout
+            return absl_container(ref)
+    else:
+        std_unordered_map(ref)
 
```

Comment:
I saw the scylla-gdb failures. How did it work until now? I see flat_hash_map class.

Maybe it's due to moving to abseil 2024?

Reply:
@avikivity yes, that was the reason. and i just fixed it in the latest revision.